stat_test.go

```diff
@@ -79,6 +79,9 @@ func TestCorrelation(t *testing.T) {
 			t.Errorf("Correlation mismatch case %d. Expected %v, Found %v", i, test.ans, c)
 		}
 	}
+	if !Panics(func() { Correlation(make([]float64, 2), make([]float64, 3), make([]float64, 3)) }) {
```

Comment:
Given you're adding this, could you also add two more tests, one where `weights = nil` and one where `len(x) = len(y) != len(w)`. Thanks.